ts/components/dialog/OnionStatusPathDialog.tsx

```diff
@@ -65,6 +66,15 @@ const OnionPathModalInner = () => {
     },
   ];
 
+  // Ensure we can always find the GeoLite2 database, regardless of whether
+  // this is a dev or a prod build.
+  const binPath = (process.env.NODE_APP_INSTANCE || '').startsWith('devprod')
+    ? path.resolve(`${__dirname}/../../..`)
+    : path.resolve(`${process.resourcesPath}/..`);
+  const buffer = readFileSync(`${binPath}/mmdb/GeoLite2-Country.mmdb`);
+  const reader = new Reader<CountryResponse>(buffer);
+  const lang = 'en';
+
```

Comment:
yeah, this cannot be made in the render thread, and especiially not here, this readFileSync called will be made every single time this React components rerenders. 

I would advise to move that readFileSync and the reader creation in `preload.js` ( you can have a look at how the same thing is done for `getSeedNodeList`

Reply:
I'm struggling with moving this to `preload.js`.

It all goes well until:

`const reader = new Reader<CountryResponse>(buffer);`

With that in `preload.js`, the client never gets past the splash screen. It's as if it's not an asynchronous operation when it's done at this stage, but I'm guessing.

Any ideas?